### PR TITLE
Attempt supporting negative voltage

### PR DESCRIPTION
**Pull request by lashtear**
cherry-picked for jrddunbr's fork

### CODE REVIEW

**Comment by lashtear** [comment]
Been working with it for a while, but check with @OmegaHaxors; they had more AC designs to use as validation.

In general, conditions with negative voltage (relative to common ground) previously would cause cables to explode, where with this they see it exactly as they would positive voltage, but in the opposite polarity.  This has many applications in circuit design, but I have not tested what other weaknesses in the model it might expose.  (Grant you without this, they just blow up immediately...)

**Comment by jrddunbr** [comment]
I'll fiddle with this later. If it doesn't crash stuff, I'll merge it.

**Comment by jrddunbr** [comment]
Hmmm. I think we are making some *bad* assumptions here.

**Comment by jrddunbr** [comment]
Look at the power output on that cable. There's something wrong there :D

**Comment by OmegaHaxors** [comment]
In regards to the wire voltage, yeah. It just takes Voltage and multiples it by Current, it's not at all realistic to  what actual amount of power is being put through, but it can be used to approximate. An actual useful number would be calculating the amount of power lost in the particular target component, since that will always be consistent no matter what the current or voltage is.

TL;DR it's an artifact of the way the meter measures current. As far as the engine's concerned, nothing's invalid here.

**Comment by jrddunbr** [comment]
approved
